internal/storage/v1/kafka/writer.go

```diff
@@ -21,14 +22,14 @@ type spanWriterMetrics struct {
 // SpanWriter writes spans to kafka. Implements spanstore.Writer
 type SpanWriter struct {
 	metrics    spanWriterMetrics
-	producer   sarama.AsyncProducer
+	producer   tracestore.Writer
 	marshaller Marshaller
 	topic      string
 }
 
 // NewSpanWriter initiates and returns a new kafka spanwriter
 func NewSpanWriter(
-	producer sarama.AsyncProducer,
+	producer tracestore.Writer,
```

Comment:
there is no Kafka-specific implementation of `tracestore.Writer`. You need to instantiate & start kafka exporter from OTEL and use that to write data.